lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -249,24 +246,53 @@ public APIGatewayProxyResponseEvent handleRequest(
             LogHelper.logErrorMessage("Failed to parse request body into map.", e.getMessage());
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_IP_ADDRESS);
-        } catch (VerifiableCredentialException e) {
-            LogHelper.logErrorMessage(
-                    "Inherited identity validation failed.", e.getErrorResponse().getMessage());
-            return ApiGatewayResponseGenerator.proxyJsonResponse(
-                    HttpStatus.SC_BAD_REQUEST, ErrorResponse.INVALID_SESSION_REQUEST);
         }
     }
 
-    private void validateAndStoreHMRCInheritedIdentity(String inheritedIdentityJWT, String userId)
-            throws ParseException, VerifiableCredentialException {
-        SignedJWT signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJWT);
-        CriConfig inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
+    @Tracing
+    private void validateAndStoreHMRCInheritedIdentity(String userId, JWTClaimsSet claimsSet)
+            throws RecoverableJarValidationException, ParseException {
+        SignedJWT signedInheritedIdentityJWT;
+        try {
+            signedInheritedIdentityJWT =
+                    SignedJWT.parse(
+                            ((JarClaims) claimsSet.getClaim("claims"))
```

Comment:
I'm not convinced this will work - I think you need to explicitly map this as a JSON value (you can't just cast it).

Reply:
Yep, you're quite right. The tests are all constructing a claimsSet with the right objects in which is why the tests pass. Need to make sure it's been deserialized/serialized to be a valid test.